pkg/controller/gather_commands.go

```diff
@@ -271,3 +289,56 @@ func record(functionReports []gather.GathererFunctionReport,
 	}
 	return recdriver.LastArchive()
 }
+
+// dataStatus is a helper struct to unmarshall
+// the HTTP response from the processing status endpoint
+type dataStatus struct {
+	ClusterID string `json:"cluster"`
+	Status    string `json:"status"`
+}
+
+// wasDataProcessed polls the "insights-results-aggregator" service processing status endpoint using provided
+// "insightsRequestID" and tries to parse the response body in case of HTTP 200 response.
+func wasDataProcessed(ctx context.Context,
+	insightsCli processingStatusClient,
+	insightsRequestID string, controllerConf *config.Controller) (bool, error) {
+	delay := controllerConf.ReportPullingDelay
+	retryCounter := 0
+	klog.V(4).Infof("Initial delay when checking processing status: %v", delay)
+
+	var resp *http.Response
+	err := wait.PollUntilContextCancel(ctx, delay, false, func(ctx context.Context) (done bool, err error) {
+		resp, err = insightsCli.GetDataProcessingStatus(ctx, // nolint: bodyclose
+			controllerConf.ProcessingStatusEndpoint, insightsRequestID) // response body is closed later
+		if err != nil {
+			return false, err
+		}
+		if resp.StatusCode == http.StatusOK || retryCounter == 2 {
+			return true, nil
```

Comment:
I'm a bit confused.
Why if we reach 2 retries (maximum allowed retries?) but no statusOK from the request, we return "true"?

Reply:
Yes. Note that this return value is for the `wait.ConditionWithContextFunc` and not for the `wasDataProcessed` function and it means that we don't want to poll the endpoint again - i.e the condition was satisfied.